app/services/user_event_creator.rb

```diff
@@ -79,18 +79,11 @@ def build_disavowal_token
   # @return [Array(Event, String)] an (event, disavowal_token) tuple
   def create_event_for_new_device(event_type:, user:, disavowal_token:)
```

Comment:
It didn't previously occur to me that the way user events work is very generalized to sending a notification for _any_ event for _any_ unrecognized device, whereas what we're building is very specific to signing in. On the other hand, the "generic" logic has always [sent the very-specific `new_device_sign_in` mailer template](https://github.com/18F/identity-idp/blob/bf4f7c16a9899d615abfca0829b56b27c994ee14/app/services/user_alerts/alert_user_about_new_device.rb#L15), so maybe we had some false assumptions about how generic this truly is.

It might be worth revisiting whether this class should bother dealing with "existing" vs. "new" devices, or if instead we should move the new device handling to code related to the sign-in flow.

Reply:
Related Slack discussion: https://gsa-tts.slack.com/archives/C0NGESUN5/p1712598516686569

I think there's a potential refactor here, though after time-boxing an attempt at it, it generally amounted to shifting complexity around, but not actually meaningfully reducing it. I don't think the side-effect nature of this is _a bad thing_ necessarily, but I wish there was a stronger link between the side-effect and the resulting notification.